docs: add optional field in secretKeyRef (#1590)
keptn/lifecycle-toolkit: docs/content/en/docs/yaml-crd-ref/metricsprovider.md

  secretKeyRef:
    name: <secret-name>
    key: <secret-key-that-holds-token>

>     optional: true

Reviewer: ```suggestion
    optional: true | false
```
This is the synopsis so you need to show all allowable values

Author: Ok but how can i show previous table in this

  * **secretKeyRef**
    * **name:** -- Name of the token for this data provider
    * **key:** -- Key for this data provider

>     * **optional** -- allows you to specify whether the secret key is required or not

Reviewer: ```suggestion
    * **optional** -- Boolean value that specifies whether the secret `key` is required
```

>     * **optional** -- Boolean value that specifies whether the secret `key` is required

Reviewer 2: I suggest using the same documentation text that k8s uses:
```suggestion
    * **optional** -- Specify whether the Secret or its key must be defined
```

source: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#secretkeyselector-v1-core

Author: done
